executor/internal/mpp/local_mpp_coordinator.go

```diff
@@ -394,9 +451,20 @@ func (c *localMppCoordinator) handleDispatchReq(ctx context.Context, bo *backoff
 // NOTE: We do not retry here, because retry is helpless when errors result from TiFlash or Network. If errors occur, the execution on TiFlash will finally stop after some minutes.
 // This function is exclusively called, and only the first call succeeds sending Tasks and setting all Tasks as cancelled, while others will not work.
 func (c *localMppCoordinator) cancelMppTasks() {
+	usedStoreAddrs := make(map[string]bool)
 	c.mu.Lock()
-	defer c.mu.Unlock()
-	c.sessionCtx.GetMPPClient().CancelMPPTasks(kv.CancelMPPTasksParam{Reqs: c.mppReqs})
+	for _, task := range c.mppReqs {
```

Comment:
Maybe add some notes?
- The mpptask of the same addr only needs to send a cancel request, because of xxxxx
- If one mpptask is canceled, other mpp tasks must also be canceled

Reply:
Make sense

Reply:
Done